src/explore-education-statistics-frontend/src/modules/table-tool/TableToolPage.tsx

```diff
@@ -104,6 +110,13 @@ class TableToolPage extends Component<Props, State> {
     timePeriodRange: [],
     tableData: [],
     footnotes: [],
+    tableHeaders: {
+      columnGroups: [],
+      columns: [],
+      rowGroups: [],
+      rows: [],
+    },
+    dataTableRef: React.createRef(),
```

Comment:
Refs shouldn't go on component state as they are mutable and their changes aren't really tracked by React. They should be assigned as class properties.